webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -613,7 +614,31 @@ class PolicyWriterServiceImpl(
     res
   }
 
-  // just write an empty file for now
+
+  private[this] def writeDirectiveCsv(paths: NodePromisesPaths, policies: Seq[Policy], policyMode : GlobalPolicyMode) =  {
+    val path = new File(paths.newFolder, "rudder-directives.csv")
+
+    val csvContent = for {
+      policy <- policies
+    } yield {
+      ( policy.id.directiveId.value ::
+      policy.policyMode.getOrElse(policyMode.mode).name ::
+      policy.technique.generationMode.name ::
+      policy.technique.agentConfig.runHooks.nonEmpty ::
+      policy.technique.id.name ::
+      policy.technique.id.version ::
+      policy.directiveOrder.value ::
+      Nil ).mkString("\"","\",\"","\"")
+
```

Comment:
I believe we need the list to be sorted by `directiveOrder` and that we need to know if the technique is system or not.